client/blocks/plan-storage/index.jsx

```diff
@@ -2,51 +2,47 @@
  * External dependencies
  */
 import classNames from 'classnames';
-import React from 'react';
+import React, { Component, PropTypes } from 'react';
 import { connect } from 'react-redux';
-import noop from 'lodash/noop';
 
 /**
  * Internal dependencies
  */
 import QueryMediaStorage from 'components/data/query-media-storage';
 import { getMediaStorage } from 'state/sites/media-storage/selectors';
 import { getSite } from 'state/sites/selectors';
-import PlanStorageButton from 'my-sites/plan-storage/button';
+import PlanStorageBar from './bar';
 
-const PlanStorage = React.createClass( {
-
-	propTypes: {
-		className: React.PropTypes.string,
-		mediaStorage: React.PropTypes.object,
-		siteId: React.PropTypes.number.isRequired,
-		onClick: React.PropTypes.func
-	},
-
-	getDefaultProps() {
-		return {
-			onClick: noop
-		}
-	},
+class PlanStorage extends Component {
+	static propTypes = {
+		className: PropTypes.string,
+		mediaStorage: PropTypes.object,
+		siteId: PropTypes.number.isRequired,
+	};
 
 	render() {
-		if ( ! this.props.site || this.props.site.jetpack ) {
+		const { site, siteId } = this.props;
+
+		if ( ! site || site.jetpack ) {
```

Comment:
Let's use more specific selectors, and pass those values through in props. For example: `isJetpackSite`, `getSiteSlug`